### DISCUSSION TITLE
Split out separate prop from `autoResetPageIndex` for editing

### DISCUSSION THREAD

**Discussion by imnasnainaec** [KevinVandy/material-react-table#1157] [Ideas]
It is nice to have `autoResetPageIndex` set to `true` (the default) for filtering/sorting/grouping, yet at the same time it would be nice to set it to `false` just for row edits / data updates. We have tables with thousands of editable rows and cannot have it jumping back to the first page every time somebody edits a single row. Is it possible to split `autoResetPageIndex` into two separate props: one for filtering/sorting/grouping and one for editing?

**Comment by kennsorr** [comment]
You can set `autoResetPageIndex` to 'false' and just use `table.setPageIndex()` to update the page index. You can console log the `table` object to get a better idea of the available methods for changing table state. `table.setState()` is useful for adding custom behavior.

`const table = useMaterialReactTable({ ... })`

`console.log(table)`

Would be great if this was in the documentation! At least I didn't see it

**Reply by KevinVandy** [reply]
There's an entire docs page for this: https://www.material-react-table.com/docs/api/table-instance-apis

Not to mention all the the TanStack Table docs

**Reply by imnasnainaec** [reply]
That's helpful for explicitly going back to the first page, thank you, @kennsorr. However, that doesn't resolve the timing of doing so. If I want to use `table.setPageIndex()` when the filtering changes, I could use `onColumnFiltersChange`, but then I have to manually handle all the filtering myself, which would be a shame when the default table filtering functionality is doing such a great job.

**Reply by KevinVandy** [reply]
https://tanstack.com/table/latest/docs/faq#how-do-i-stop-my-table-state-from-automatically-resetting-when-my-data-changes

Maybe that helps?

**Reply by kennsorr** [reply]
Yea you'd have to just reproduce the default sorting behavior and add any custom behavior in there as well.

```
  const table = useMaterialReactTable({
    onColumnFiltersChange:  (updaterOrValue: any) => {
      
      // Custom behavior (i.e. reset page index)

      // resume default
      const newFilterState =
        typeof updaterOrValue === 'function'
          ? updaterOrValue(table.getState().columnFilters)
          : updaterOrValue
  
      table.setState((prevState) => ({
        ...prevState,
        columnFilters: newFilterState
      }))
    },
  });
```

**Reply by KevinVandy** [reply]
I don't know how we would separate the behavior. It just looks for when the reference of the data changes.

**Reply by imnasnainaec** [reply]
@KevinVandy That's slick, thank you.
